### DISCUSSION TITLE
action does not use cached layers when rebuilding image

### DISCUSSION THREAD

**Discussion by johanneszellinger** [docker/build-push-action#943] [Q&A]
Trying to used the following step in a GitHub workflow to build an image on an Enterprise server:

      - name: Build image
        uses: docker/build-push-action@v4
        with:
          context: .
          file: ./src/Dockerfile
          push: true
          tags: ${{ env.REGISTRY }}/${{ env.REGISTRY_PATH }}/image:latest
          cache-from: type=gha
          cache-to: type=gha, mode=max

Running the workflow and building the image works fine, however I would expect that the build layers are cached on the GitHub server and re-running the same workflow without changing the Dockerfile should be faster and also show `CACHED` in the unchanged steps. 

I can see that the layers are actually cached on the server:

[![enter image description here][1]][1]

However, it does not seem, that the caches are used. For example the base image is `python:3.11` which gets downloaded every time (although this should be the cache with 190MB in the image above I think). 

Searching the action logs for any "cache" entry only shows:

    #12 importing cache manifest from gha:14508570815286874220
    #12 DONE 0.0s

What might be the reason the caches are not used for rebuilding the image?

  [1]: https://i.stack.imgur.com/CMavW.png

**Comment by igor47** [comment]
I'm having the same issue. I have a very full cache, that never seems to be used.
